docs/docs/api-reference/core/selector.md

```diff
@@ -3,19 +3,47 @@ title: selector(options)
 sidebar_label: selector()
 ---
 
-Returns writeable or read-only Recoil state, depending on the options passed to the function.
-
-Selectors represent **derived state**. You can think of derived state as the output of passing state to a pure function that modifies the given state in some way.
+*Selectors* represent a function, or **derived state** in Recoil  You can think of them as a "pure function" without side-effects that always returns the same value for a given set of dependency values.  If only a `get` function is provided the selector is read-only and returns a `RecoilValueReadOnly` object.  If a `set` is also provided it returns a writeable `RecoilState` object.
 
 ---
 
-- `options`
-  - `key`: A unique string used to identify the atom internally. This string should be unique with respect to other atoms and selectors in the entire application.
-  - `get`: A function that is passed an object as the first parameter containing the following properties:
-    - `get`: a function used to retrieve values from other atoms/selectors. All atoms/selectors passed to this function will be implicitly added to a list of **dependencies** for the selector. If any of the selector's dependencies change, the selector will re-evaluate.
-  - `set?`: If this property is set, the selector will return **writeable** state. A function that is passed an object as the first parameter containing the following properties:
-    - `get`: a function used to retrieve values from other atoms/selectors. This function will not subscribe the selector to the given atoms/selectors.
-    - `set`: a function used to set the values of Recoil state. The first parameter is the Recoil state and the second parameter is the new value.
+```jsx
+function selector<T>({
+  key: string,
+
+  get: ({
+    get: GetRecoilValue
+  }) => T | Promise<T> | RecoilValue<T>,
+
+  set?: (
+    {
+      get: GetRecoilValue,
+      set: SetRecoilValue,
+      reest: ResetRecoilValue,
```

Comment:
hey @drarmstr, slight typo here fyi `reest` instead of `reset`

Reply:
Thanks, that was fixed in #275